AEPServices/Mocks/PublicTestUtils/NetworkRequestHelper.swift

```diff
@@ -100,12 +83,9 @@ class NetworkRequestHelper {
     ///
     /// - Returns: A `DispatchTimeoutResult` with the result of the wait operation, or `nil` if the `NetworkRequest` does not match any expected request.
     private func awaitFor(networkRequest: NetworkRequest, timeout: TimeInterval) -> DispatchTimeoutResult? {
-        queue.sync {
-            for expectedNetworkRequest in expectedNetworkRequests where expectedNetworkRequest.key == TestableNetworkRequest(from: networkRequest) {
-                return expectedNetworkRequest.value.await(timeout: timeout)
-            }
-
-            return nil
+        let testableNetworkRequest = TestableNetworkRequest(from: networkRequest)
+        return queue.sync {
+            return self.expectedNetworkRequests[testableNetworkRequest]?.await(timeout: timeout)
```

Comment:
Can you use the queue to only query `expectedNetworkRequests` and move the `await` outside ?